### PR TITLE
Declarative pipeline

### PR DESCRIPTION
**Pull request by stephendonner**
@davehunt r?

Addresses https://github.com/mozilla/fxapom/issues/77

Passing ad-hoc build: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/fxapom.adhoc/46/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,154 +1,65 @@
...
-          }
+    stage('Test') {
+      steps {
+        writeCapabilities(capabilities, 'capabilities.json')
+        sh "tox -e py27"
+      }
+      post {
+        always {
+          junit 'results/*.xml'
+          archiveArtifacts 'results/*'
```
Comment (bitwiseman) [inline Jenkinsfile]: Minor point: Suggest you do `archiveArtifacts` first.  That way you are sure to get the files in results, even if junit step fails.

Comment (stephendonner) [inline Jenkinsfile]: Fixed; thanks!

File: Jenkinsfile [hunk 2]
```diff
@@ -1,154 +1,65 @@
...
+    failure {
+      mail(
+        body: "${BUILD_URL}",
+        from: "firefox-test-engineering@mozilla.com",
+        replyTo: "firefox-test-engineering@mozilla.com",
+        subject: "Build failed in Jenkins: ${JOB_NAME} #${BUILD_NUMBER}",
+        to: "fte-ci@mozilla.com")
+    }
+    changed {
+      ircNotification('#fx-test-alerts')
```
Comment (davehunt) [inline Jenkinsfile]: This is the default channel, so it can be omitted.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,154 +1,65 @@
...
+      post {
+        always {
+          archiveArtifacts 'results/*'
+          junit 'results/*.xml'
+          publishHTML(target: [
+            allowMissing: false,
+            alwaysLinkToLastBuild: true,
+            keepAll: true,
+            reportDir: 'results',
+            reportFiles: "tests.html",
```
Comment (davehunt) [inline Jenkinsfile]: The environment is `py27` so the HTML report should be `py27.html`.

File: Jenkinsfile [hunk 4]
```diff
@@ -1,154 +1,65 @@
...
-        sleep 5
-        echo "JOIN ${channel}"
-        echo "NOTICE ${channel} :${message}"
-        echo QUIT
-        ) | openssl s_client -connect irc.mozilla.org:6697
-    """
+  environment {
+    PYTEST_ADDOPTS =
+      "-n=10 " +
+      "--color=yes " +
```
Comment (davehunt) [inline Jenkinsfile]: Please also add `--tb=short` as we no longer inherit from the global environment.

File: Jenkinsfile (hunk 3 above)
Comment (davehunt) [inline Jenkinsfile]: I'm not sure why this didn't cause the build to fail...

File: Jenkinsfile (hunk 4 above)
Comment (stephendonner) [inline Jenkinsfile]: Done!

File: Jenkinsfile [hunk 5]
```diff
@@ -1,154 +1,66 @@
...
+      "--tb=short" +
```
Comment (stephendonner) [inline Jenkinsfile]: Fixed, I hope!

**Pull request merged**